src/utils/hooks/use-animation.ts

```diff
@@ -0,0 +1,53 @@
+import { useRef, useEffect } from 'react'
+import { Animated } from 'react-native'
+
+export const useAnimatedValue = (initialValue: number): Animated.Value => {
+  const ref = useRef(new Animated.Value(initialValue))
+  return ref.current
+}
+
+interface BaseAnimationConfig {
+  initialValue?: number
+}
+
+export type TimingAnimationConfig = BaseAnimationConfig &
+  ({ type: 'timing' } & Animated.TimingAnimationConfig)
+
+export type SpringAnimationConfig = BaseAnimationConfig &
+  ({ type: 'spring' } & Animated.SpringAnimationConfig)
```

Comment:
Should these two types be exported? Considering `UseAnimationConfig` is the one actually used

Reply:
I think so, as otherwise, you'll have some difficulty using the `UseAnimationConfig` type if you try using it with `Pick`, `Required` and other "type-helpers" like that. Furthermore, since these types represent the front-facing API, I would leave them there, especially since they don't add any code at runtime (as they are just types).